Read australian_users_items.json that contains values in pyhton dictionary format. Filter out the values by fetching user_id, item_id, item_name and playtime_forever and save it in csv file output_australian_users_items.csv

In [3]:
import csv
import ast
import json
import pandas as pd

csv_columns = ['user_id','item_id','item_name','playtime_forever']

with open('C:\\Users\\Kriti\\Desktop\\CMPE256\\CMPE256_Project\\australian_users_items.json', encoding='utf-8') as data_file:
    
    with open('C:\\Users\\Kriti\\Desktop\\CMPE256\\CMPE256_Project\\output_australian_users_items.csv', 'w', encoding='utf-8', newline='') as csv_file:
        line = data_file.readline()
        writer = csv.DictWriter(csv_file, fieldnames=csv_columns)
        writer.writeheader()
        while line:
            json_data = (ast.literal_eval(line))
            line = data_file.readline()
            for item in json_data["items"]:
                new_dict_data={'user_id':json_data["user_id"],'item_id':item['item_id'],'item_name':item['item_name'],'playtime_forever':item['playtime_forever']}
            
                writer.writerow(new_dict_data)
                line = data_file.readline()

Read output_australian_users_items.csv  file and convert it in data frame users_items_df

In [4]:
print(__doc__)

import numpy as np
import scipy as sp
import pandas as pd

from sklearn.preprocessing import Imputer

#read the csv into a dataframe
users_items_df = pd.read_csv('C:\\Users\\Kriti\\Desktop\\CMPE256\\CMPE256_Project\\output_australian_users_items.csv')


Automatically created module for IPython interactive environment


In [6]:
users_items_df.head(50)

user_id  item_id                              item_name  \
0   76561197970982479       10                         Counter-Strike   
1   76561197970982479       20                  Team Fortress Classic   
2   76561197970982479       30                          Day of Defeat   
3   76561197970982479       40                     Deathmatch Classic   
4   76561197970982479       50              Half-Life: Opposing Force   
5   76561197970982479       60                               Ricochet   
6   76561197970982479       70                              Half-Life   
7   76561197970982479      130                  Half-Life: Blue Shift   
8   76561197970982479      300                  Day of Defeat: Source   
9   76561197970982479      240                 Counter-Strike: Source   
10  76561197970982479     3830                            Psychonauts   
11  76561197970982479     2630                         Call of Duty 2   
12  76561197970982479     3900            Sid Meier's Civilization IV   
13  76561197970982479    34440            Sid Meier's Civilization IV   
14  76561197970982479     3920                   Sid Meier's Pirates!   
15  76561197970982479     6400                       Joint Task Force   
16  76561197970982479     6910      Deus Ex: Game of the Year Edition   
17  76561197970982479     7670                               BioShock   
18  76561197970982479   409710                    BioShock Remastered   
19  76561197970982479      220                            Half-Life 2   
20  76561197970982479      320                Half-Life 2: Deathmatch   
21  76561197970982479      340                Half-Life 2: Lost Coast   
22  76561197970982479      360           Half-Life Deathmatch: Source   
23  76561197970982479      380               Half-Life 2: Episode One   
24  76561197970982479      400                                 Portal   
25  76561197970982479      420               Half-Life 2: Episode Two   
26  76561197970982479     9340     Company of Heroes: Opposing Fronts   
27  76561197970982479   228200  Company of Heroes (New Steam Version)   
28  76561197970982479    11450                               Overlord   
29  76561197970982479     7940         Call of Duty 4: Modern Warfare   
30  76561197970982479     4700                 Medieval II: Total War   
31  76561197970982479    12900                              Audiosurf   
32  76561197970982479    13250                            Unreal Gold   
33  76561197970982479    16100          Virtual Villagers: A New Home   
34  76561197970982479    15700                Oddworld: Abe's Oddysee   
35  76561197970982479    15710                Oddworld: Abe's Exoddus   
36  76561197970982479    17330                         Crysis Warhead   
37  76561197970982479    17340                            Crysis Wars   
38  76561197970982479    22000                           World of Goo   
39  76561197970982479      500                            Left 4 Dead   
40  76561197970982479     4560                      Company of Heroes   
41  76561197970982479    17460                            Mass Effect   
42  76561197970982479    10500                      Empire: Total War   
43  76561197970982479    24740     Burnout Paradise: The Ultimate Box   
44  76561197970982479    22200                             Zeno Clash   
45  76561197970982479    26800                                  Braid   
46  76561197970982479     1250                          Killing Floor   
47  76561197970982479    35420  Killing Floor Mod: Defence Alliance 2   
48  76561197970982479     3590   Plants vs. Zombies: Game of the Year   
49  76561197970982479     8880                          Freedom Force   

    playtime_forever  
0                  6  
1                  0  
2                  7  
3                  0  
4                  0  
5                  0  
6                  0  
7                  0  
8               4733  
9               1853  
10               333  
11            

Unique count of user_id and item_id=85876

In [16]:
users_items_df.groupby(["user_id","item_id"]).size().count()

85876

Unique count of user_id=1796

In [17]:
users_items_df.groupby(["user_id"]).size().count()

1796

Unique count of item_id=5675

In [18]:
users_items_df.groupby(["item_id"]).size().count()

5675

In [19]:
users_items_df.describe()

item_id  playtime_forever
count   85876.000000      85876.000000
mean   179230.012704       1059.054055
std    130409.849863       5443.709166
min        10.000000          0.000000
25%     35420.000000          0.000000
50%    214560.000000         36.000000
75%    265930.000000        404.000000
max    529670.000000     337453.000000

Read australian_user_reviews.json that contains values in pyhton dictionary format. Filter out the values by fetching user_id, item_id, recommend and save it in csv file output_australian_user_reviews.csv

In [20]:
import csv
import ast
import json
import pandas as pd

csv_columns = ['user_id','item_id','recommend']

with open('C:\\Users\\Kriti\\Desktop\\CMPE256\\CMPE256_Project\\australian_user_reviews.json', encoding='utf-8') as data_file:
    
    with open('C:\\Users\\Kriti\\Desktop\\CMPE256\\CMPE256_Project\\output_australian_user_reviews.csv', 'w', encoding='utf-8', newline='') as csv_file:
        line = data_file.readline()
        writer = csv.DictWriter(csv_file, fieldnames=csv_columns)
        writer.writeheader()
        while line:
            json_data = (ast.literal_eval(line))
            line = data_file.readline()
            for review in json_data["reviews"]:
                new_dict_data={'user_id':json_data["user_id"],'item_id':review['item_id'],'recommend':review['recommend']}
            
                writer.writerow(new_dict_data)
                line = data_file.readline()



Read output_australian_user_reviews.csv  file and convert it in data frame user_reviews_df

In [21]:
print(__doc__)

import numpy as np
import scipy as sp
import pandas as pd

from sklearn.preprocessing import Imputer

#read the csv into a dataframe
user_reviews_df = pd.read_csv('C:\\Users\\Kriti\\Desktop\\CMPE256\\CMPE256_Project\\output_australian_user_reviews.csv')


Automatically created module for IPython interactive environment


In [22]:
user_reviews_df.head(50)

user_id  item_id  recommend
0       76561197970982479     1250       True
1       76561197970982479    22200       True
2       76561197970982479    43110       True
3               maplemage   211420       True
4               maplemage   211820       True
5               maplemage      730       True
6               maplemage   204300       True
7       76561198156664158   252950       True
8              WeiEDKrSat     4000       True
9            death-hunter   232090       True
10           death-hunter     1250       True
11           death-hunter     4000       True
12           MarbleShrine   220200       True
13               devvonst      440       True
14               devvonst   218740       True
15               devvonst      550       True
16      76561198048353577   368230       True
17      76561198048353577     4000       True
18      76561198048353577    33930       True
19             SavageEeEe   221100       True
20          KanadeTheNeko   331470       True
21          KanadeTheNeko     4000       True
22      76561198054540475    12210       True
23      76561198054540475   265210       True
24      76561198054540475    48700       True
25                boydeer      440       True
26                boydeer   383080      False
27      76561198075277292      730       True
28      76561198075277292      570       True
29              diego9031     4000       True
30              diego9031   317360       True
31              diego9031   223710       True
32              diego9031   299360       True
33              diego9031   322330       True
34              diego9031     1250       True
35              diego9031    12100       True
36              diego9031   204360       True
37              diego9031      440       True
38              diego9031   224260       True
39      76561198138053270   218620       True
40            Life-Of-Blu   377160       True
41            Life-Of-Blu   211820       True
42           TheDropix101   242920       True
43  iamthekingofbrowntown   344760      False
44  iamthekingofbrowntown   252950       True
45  iamthekingofbrowntown   236430       True
46  iamthekingofbrowntown   399730       True
47           Mossy_Gaming      440       True
48      76561198087407319      440       True
49             sad-commie   294100       True

Recommed values are true or false. Assuming if user recommends item i.e. value is true that means he likes the item and if false that means he dislikes the item. 

In [24]:
user_reviews_df["recommend"].unique()

array([ True, False])

In [25]:
user_reviews_df.describe()

item_id
count   17950.000000
mean   166974.360501
std    131685.907734
min        10.000000
25%      8870.000000
50%    214950.000000
75%    259080.000000
max    521990.000000

Unique values are 17904 and total count is 17950. So df contains duplicate values

In [29]:
user_reviews_df.groupby(["user_id","item_id","recommend"]).size().count()

17904

Remove duplicate values from dataframe

In [30]:
user_reviews_df.drop_duplicates()

user_id  item_id  recommend
0         76561197970982479     1250       True
1         76561197970982479    22200       True
2         76561197970982479    43110       True
3                 maplemage   211420       True
4                 maplemage   211820       True
5                 maplemage      730       True
6                 maplemage   204300       True
7         76561198156664158   252950       True
8                WeiEDKrSat     4000       True
9              death-hunter   232090       True
10             death-hunter     1250       True
11             death-hunter     4000       True
12             MarbleShrine   220200       True
13                 devvonst      440       True
14                 devvonst   218740       True
15                 devvonst      550       True
16        76561198048353577   368230       True
17        76561198048353577     4000       True
18        76561198048353577    33930       True
19               SavageEeEe   221100       True
20            KanadeTheNeko   331470       True
21            KanadeTheNeko     4000       True
22        76561198054540475    12210       True
23        76561198054540475   265210       True
24        76561198054540475    48700       True
25                  boydeer      440       True
26                  boydeer   383080      False
27        76561198075277292      730       True
28        76561198075277292      570       True
29                diego9031     4000       True
...                     ...      ...        ...
17920     76561198232478272      440       True
17921     76561198236893796   383580       True
17922     76561198236893796      440       True
17923     76561198236893796      730       True
17924     76561198236893796   391460      False
17925     76561198236893796   209000       True
17926               wayfeng      730       True
17927               wayfeng   369200       True
17928               wayfeng     4000      False
17929           kimjongadam   224480       True
17930           kimjongadam   314160       True
17931           kimjongadam   273110      False
17932           kimjongadam   264200       True
17933           kimjongadam   239820       True
17934           SenpaiDemon      730       True
17935  Fuckfhaisjnsnsjakaka      730       True
17936          ChrisCoroner   209120       True
17937     76561198267374962   369200       True
17938     76561198270958927   381210       True
17939     76561198272389051      440       True
17940     76561198277602337   306130      False
17941            vinquility   431510       True
17942            vinquility   413150       True
17943               sexyawp   427730       True
17944        JustMielThings      570       True
17945              Ghoustik      730       True
17946     76561198312638244   233270       True
17947     76561198312638244      130       True
17948     76561198312638244       70       True
17949     76561198312638244   362890       True

[17904 rows x 3 columns]

Merge user_items and user_reviews on item_id and user_id 

In [31]:
merged_df = pd.merge(user_reviews_df, users_items_df, on=['item_id','user_id'], how='outer')
merged_df.head()

user_id  item_id recommend      item_name  playtime_forever
0  76561197970982479     1250      True  Killing Floor           10006.0
1  76561197970982479    22200      True     Zeno Clash             271.0
2  76561197970982479    43110      True     Metro 2033             834.0
3          maplemage   211420      True            NaN               NaN
4          maplemage   211820      True            NaN               NaN

In [40]:
user_reviews_df.groupby(["user_id"]).size().count()

7823

In [41]:
user_reviews_df.groupby(["item_id"]).size().count()

2167

There are many user items combination present in user_reviews_df that are not present in users_items_df

In [42]:
merged_df.groupby(["user_id","item_id"]).size().count()

103529

merged_df has data type object and now has nan values

In [43]:
merged_df["recommend"].unique()

array([True, False, nan], dtype=object)

Importing bundle data into csv file

In [44]:
import csv
import ast
import json
import pandas as pd

import csv
import ast
import json
import pandas as pd

csv_columns = ['bundle_id','item_id','genre','discounted_price','bundle_final_price','bundle_price','bundle_name']

with open('C:\\Users\\Kriti\\Desktop\\CMPE256\\CMPE256_Project\\bundle_data.json', encoding='utf-8') as data_file:
    
    with open('C:\\Users\\Kriti\\Desktop\\CMPE256\\CMPE256_Project\\output_bundle_data.csv', 'w', encoding='utf-8', newline='') as csv_file:
        line = data_file.readline()
        writer = csv.DictWriter(csv_file, fieldnames=csv_columns)
        writer.writeheader()
        while line:
            json_data = (ast.literal_eval(line))
            line = data_file.readline()
            for item in json_data["items"]:
                new_dict_data={'bundle_id':json_data["bundle_id"],'item_id':item['item_id'],'genre':item['genre'],'discounted_price':item['discounted_price'],'bundle_final_price':json_data["bundle_final_price"],'bundle_price':json_data["bundle_price"],'bundle_name':json_data["bundle_name"]}
            
                writer.writerow(new_dict_data)
                line = data_file.readline()



In [45]:
print(__doc__)

import numpy as np
import scipy as sp
import pandas as pd

from sklearn.preprocessing import Imputer

#read the csv into a dataframe
bundle_df = pd.read_csv('C:\\Users\\Kriti\\Desktop\\CMPE256\\CMPE256_Project\\output_bundle_data.csv')


Automatically created module for IPython interactive environment


In [47]:
bundle_df.head(50)

bundle_id item_id                                 genre discounted_price  \
0         450  326950                 Adventure, Indie, RPG            $8.99   
1         450  331490                 Adventure, Indie, RPG            $2.99   
2         450  331491                 Adventure, Indie, RPG            $1.99   
3         450  331492                 Adventure, Indie, RPG            $0.99   
4         450  348540              Adventure, Casual, Indie           $12.99   
5         450  352010              Adventure, Casual, Indie           $12.99   
6         450  390470              Adventure, Casual, Indie            $3.99   
7         450  394050              Adventure, Casual, Indie            $2.99   
8         450  398100                         Casual, Indie            $9.99   
9         450  408770                      Adventure, Indie            $6.99   
10        450  387110                 Action, Casual, Indie            $2.99   
11        450  387120                         Action, Indie            $1.99   
12        450  396610                         Action, Indie            $2.99   
13        450  396620                         Action, Indie            $0.99   
14       1455  357070                         Action, Indie            $1.99   
15       1455  366420                         Action, Indie            $0.51   
16       1455  444940                         Action, Indie            $2.99   
17       1455  515570                         Action, Indie            $2.99   
18       1455  527000                         Action, Indie            $0.59   
19        419  401190  Action, Adventure, Indie, Simulation            $0.99   
20        419  408100  Action, Adventure, Indie, Simulation            $0.99   
21        419  408101  Action, Adventure, Indie, Simulation            $0.99   
22        419  408102  Action, Adventure, Indie, Simulation            $0.99   
23        419  431190  Action, Adventure, Indie, Simulation            $0.99   
24        419  431191  Action, Adventure, Indie, Simulation            $0.99   
25        419  434940  Action, Adventure, Indie, Simulation            $0.99   
26        419  434941  Action, Adventure, Indie, Simulation            $0.99   
27        419  434942  Action, Adventure, Indie, Simulation            $0.49   
28        419  434943  Action, Adventure, Indie, Simulation            $0.99   
29        419  434944  Action, Adventure, Indie, Simulation            $0.99   
30       1418    3300                                Casual            $4.99   
31       1418    3350                                Casual            $4.99   
32       1418    3330                                Casual            $4.99   
33       1418    3390                                Casual            $4.99   
34       1418    3310                                Casual            $4.99   
35       1418    3320                                Casual            $4.99   
36       1418    3380                                Casual            $4.99   
37       1418    3420                                Casual            $4.99   
38       1418    3430                                Casual            $4.99   
39       1418    3450                                Casual            $4.99   
40       1418    3340                                Casual            $4.99   
41       1418    3410                                Casual            $4.99   
42       1418    3440                                Casual            $4.99   
43       1418    3400                                Casual            $4.99   
44       1418    3360                                Casual            $4.99   
45       1418    3460                                Casual            $4.99   
46       1418    3480                                Casual            $4.99   
47       1418    3540                                Casual            $4.99   
48       1418    3560                      Strategy, Casual            $4.99   
49  

COnverting data type od bundle_df['item_id'] to string to match it with merged_df to join to data frames

In [48]:
bundle_df['item_id'] = bundle_df['item_id'].astype('str') 

In [49]:
merged_df['item_id'] = merged_df['item_id'].astype('str') 

Merge user-item-review data frame with bundle_df based on item_id

In [50]:
merged_df_final = pd.merge(merged_df, bundle_df, on=['item_id'], how='outer')
merged_df_final.head()

user_id item_id recommend      item_name  playtime_forever  \
0  76561197970982479    1250      True  Killing Floor           10006.0   
1       death-hunter    1250      True            NaN               NaN   
2          diego9031    1250      True            NaN               NaN   
3           GamerFag    1250      True            NaN               NaN   
4          brendoboy    1250      True            NaN               NaN   

   bundle_id   genre discounted_price bundle_final_price bundle_price  \
0      612.0  Action           $19.99             $67.20      $167.78   
1      612.0  Action           $19.99             $67.20      $167.78   
2      612.0  Action           $19.99             $67.20      $167.78   
3      612.0  Action           $19.99             $67.20      $167.78   
4      612.0  Action           $19.99             $67.20      $167.78   

                          bundle_name  
0  Killing Floor 1 Complete Your Set!  
1  Killing Floor 1 Complete Your Set!  
2  Killing Floor 1 Complete Your Set!  
3  Killing Floor 1 Complete Your Set!  
4  Killing Floor 1 Complete Your Set!

Checking how many items in bundle are having review: 2030

In [53]:
merged_df_final[(merged_df_final["bundle_id"].notnull())&(merged_df_final["recommend"].notnull())]

user_id item_id recommend                      item_name  \
0      76561197970982479    1250      True                  Killing Floor   
1           death-hunter    1250      True                            NaN   
2              diego9031    1250      True                            NaN   
3               GamerFag    1250      True                            NaN   
4              brendoboy    1250      True                            NaN   
5      76561198064026657    1250      True                            NaN   
6      76561198038840599    1250      True                            NaN   
7               henrydxb    1250      True                            NaN   
8      76561198081529182    1250      True                            NaN   
9      76561198047489572    1250      True                            NaN   
10     76561198053085687    1250      True                            NaN   
11     76561198064343153    1250     False                            NaN   
12                 KRASA    1250      True                            NaN   
13                 _SNS_    1250      True                            NaN   
14     76561198094665607    1250      True                            NaN   
15     76561198025409181    1250      True                            NaN   
16     76561198088082495    1250      True                            NaN   
17     76561198038695064    1250      True                            NaN   
18     76561198051976770    1250      True                            NaN   
19       Delirious_Panda    1250      True                            NaN   
20               jvkzero    1250      True                            NaN   
21     76561198041122979    1250      True                            NaN   
22     76561198098358388    1250      True                            NaN   
23            GraytheCat    1250      True                            NaN   
24           kineticvine    1250      True                            NaN   
25     76561198067596593    1250      True                            NaN   
26                Lubnut    1250      True                            NaN   
27     76561198072339171    1250      True                            NaN   
28             koongod01    1250      True                            NaN   
29     76561198087556976    1250      True                            NaN   
...                  ...     ...       ...                            ...   
74783          Jewele485  385800      True                            NaN   
74784  76561198073970893  385800      True                            NaN   
74785         richy12879  385800      True                            NaN   
75039           Nullahan  402620      True                            NaN   
75198      EucHellscythe  258090      True                            NaN   
75763  76561197985031488  327220      True                            NaN   
75800       alyssawesome  368730      True  No Time To Explain Remastered   
76000        quillomanar  244160      True                            NaN   
76001        madengineer  244160      True                            NaN   
76002  76561198008211473  244160      True                            NaN   
76247  76561198040946765   47700     False                            NaN   
77648            mysterd   40390      True                            NaN   
77649  76561198072053388   40390     False                            NaN   
77772      truewarrior94   24790      True                            NaN   
77889        originaldog    3320      True                            NaN   
77921    padraigotreanor  420740      True                            NaN   
77930        partycatsjr  441830      True                            NaN   
77989           Pendekar  336420      True                            NaN   
78210  76561198093768306  419460     False                            NaN   
78527   AleksoSmeksoHere  342380      True                            NaN   
78528        

Removing data not present in right format

In [54]:
cleaned_data=merged_df_final[(merged_df_final.item_id != '200960,200980,200990,201000,201010') & (merged_df_final.item_id!='42170,204731,204732') & (merged_df_final.item_id!='10,80')]

In [55]:
cleaned_data[(cleaned_data["bundle_id"].notnull())&(cleaned_data["recommend"].notnull())]

user_id item_id recommend                      item_name  \
0      76561197970982479    1250      True                  Killing Floor   
1           death-hunter    1250      True                            NaN   
2              diego9031    1250      True                            NaN   
3               GamerFag    1250      True                            NaN   
4              brendoboy    1250      True                            NaN   
5      76561198064026657    1250      True                            NaN   
6      76561198038840599    1250      True                            NaN   
7               henrydxb    1250      True                            NaN   
8      76561198081529182    1250      True                            NaN   
9      76561198047489572    1250      True                            NaN   
10     76561198053085687    1250      True                            NaN   
11     76561198064343153    1250     False                            NaN   
12                 KRASA    1250      True                            NaN   
13                 _SNS_    1250      True                            NaN   
14     76561198094665607    1250      True                            NaN   
15     76561198025409181    1250      True                            NaN   
16     76561198088082495    1250      True                            NaN   
17     76561198038695064    1250      True                            NaN   
18     76561198051976770    1250      True                            NaN   
19       Delirious_Panda    1250      True                            NaN   
20               jvkzero    1250      True                            NaN   
21     76561198041122979    1250      True                            NaN   
22     76561198098358388    1250      True                            NaN   
23            GraytheCat    1250      True                            NaN   
24           kineticvine    1250      True                            NaN   
25     76561198067596593    1250      True                            NaN   
26                Lubnut    1250      True                            NaN   
27     76561198072339171    1250      True                            NaN   
28             koongod01    1250      True                            NaN   
29     76561198087556976    1250      True                            NaN   
...                  ...     ...       ...                            ...   
74783          Jewele485  385800      True                            NaN   
74784  76561198073970893  385800      True                            NaN   
74785         richy12879  385800      True                            NaN   
75039           Nullahan  402620      True                            NaN   
75198      EucHellscythe  258090      True                            NaN   
75763  76561197985031488  327220      True                            NaN   
75800       alyssawesome  368730      True  No Time To Explain Remastered   
76000        quillomanar  244160      True                            NaN   
76001        madengineer  244160      True                            NaN   
76002  76561198008211473  244160      True                            NaN   
76247  76561198040946765   47700     False                            NaN   
77648            mysterd   40390      True                            NaN   
77649  76561198072053388   40390     False                            NaN   
77772      truewarrior94   24790      True                            NaN   
77889        originaldog    3320      True                            NaN   
77921    padraigotreanor  420740      True                            NaN   
77930        partycatsjr  441830      True                            NaN   
77989           Pendekar  336420      True                            NaN   
78210  76561198093768306  419460     False                            NaN   
78527   AleksoSmeksoHere  342380      True                            NaN   
78528        